### PR TITLE
Adds in 12g flares and the flare rifle

### PR DESCRIPTION
**Pull request by shazbot194**
flare_shot
Hello Nanotrasen. I am here with quite a deal. I know how you have acquired some rather hostile asteroids that you wish to do mining on, but they are completely unlit. Well, I have a solution to your problems. Introducing the flare rifle!

The flare rifle is made from a parts kit and the module receiver already in your database. All you have to do is wrench in the barrel and screw on all the furniture in and you are good to go. For a small fee, I can add the parts kit to your autolathe, under the secured designs of course, we don't want any accidents now do we. The flare rifle is deigned to shoot a custom made shotgun shell that contains a small flare. This flare ignites as soon as it is shot, making for quite a show as it flies through the air. We will add this design for your autolathes for free with purchase of the parts kit licence.
Disclaimer; despite the flare rifle being chambered in 12g shotgun shells, it is in no way intended to be used with any shell other then the designated flare shells. You have been warned.
Secondary disclaimer: We are not responsible if all homemade shotguns no longer spawn a equally homemade shotgun shell from the ether. 
Tertiary disclaimer: Flare shells are low power shells, and won't cycle all semi-auto actions properly, needing to be manually operated when fired.

//Transmission over//

Hello, so this PR does add in shotgun shells that shoot flares. I tried to make these as balanced as possible. For starters, they cost 2.5x more then combat shotgun shells, and last anywhere from 1/20th to 1/10th the time of a normal flare.

It also adds in a new 'improvised' shotgun. This one is made from all metal. I did try to balance it around material cost, making it a more expensive then the improvised one, but being more convenient to make, while leaving the improvised shotgun untouched in it's role as a shotgun you can literally scrap together. In all, it should costs 30 sheets of metal to make the parts kit, which, at least in cargo points, makes it more expensive then the improvised shotgun. In addition, you cannot put a sling onto the flare rifle, but you can still saw it into a sawn off.

It also has some minor fixes for the improvised shotguns.

Update 29/1/18: Hello all, I have added two big features to this PR. The first is that the flare rifle can no longer shoot combat shells perfectly, exploding in your face when you do, about 40% of the time. This will shoot the shell at you, doing the full damage of the shell. But you can add some plasteel to stop this, but as normal, if there is a shell loaded when you add the sheets, you will shoot yourself with the shell. The description will change when you add the sheets, making it possible to tell if one is reinforced. The other feature is that semi-auto shotguns like the combat shotgun don't cycle themselves when loaded with flares. The bulldog still does, mostly because the code would be too much and possible drag down performance, so it will be allowed to be the only auto shotgun to fully be compatible with flares. 

I am more then happy to receive critique, and will adjust the balance if needed. 

Update 3/3/18: Hello all, so this is a rather small update. I changed the icon to be more in line with the new sprites me and merchant are making, in addition, I removed the slap crafting for the flare rifle, which both makes the code cleaner, and cuts back on the icons needed.

🆑 shazbot
Adds: Adds in the flare rifle, flare rifle parts, and flare shells
Change: Improvised shotguns no longer spawn homemade shotgun shells in their chamber when made
Fix: Improvised shotguns now display slings correctly.
/🆑

### CODE REVIEW

**Comment by FreeStylaLT** [comment]
You sure like your guns

**Comment by shazbot194** [comment]
How can you tell?

**Comment by Kyep** [comment]
Does this mean that anyone with access to a stack of metal can make themselves an improv shotgun? 
And all it takes to get lethal ammo for it is an autolathe?

**Comment by shazbot194** [comment]
Currently all it takes is a wrench and knowing where to look and you can make an improvised shotgun from the wood furniture around the station, as well as finding and or printing a reciever.

**Comment by EldritchSigma** [comment]
I mean I do like more weapons... but imagine that a certain antag will have a very hard time. Harder than it actually has due to the flash spam. And those flares cant be extinguished

**Comment by shazbot194** [comment]
I do too, that is why I made the cost so much more then normal combat shells and also why I made them so short lived. If balance is still a problem, I have a few ideas, like making combat shotguns not auto-cycle when shooting a flare shell. It is also possible to make it so the ability that snuffs all lights either make it a lot dimmer for a while, or makes it have 0 fuel, thus burning out instantly.

**Comment by Ty-Omaha** [comment]
Should make the flare gun bright orange

**Comment by KingPhilipIII** [comment]
We need more dakka.

**Comment by Necaladun** [comment]
The fixes for improv shotguns should prolly be a second PR, to keep it more atomic and as I can't see anyone objecting to them.

30 metal to make a shotgun like this is a very good deal, and I see it being particularly problematic for shadowling rounds and mass arming of the crew. In general, I'd prefer less ways for greytide and cargo to get their hands on weapons, as this makes cracking open crates a hell of a lot easier.

**Comment by shazbot194** [comment]
I could always adjust the price to make it harder to obtain, currently it's 30 metal for the parts and 7.5 for the reciever, I could make it 43.5 metal for the parts and make the total cost exactly 50 sheets, add in something such as some sheets of plasteel, or just about anything, I am open to suggestions.

**Comment by Necaladun** [comment]
Adding some plasteel, and maybe something a bit rarer would be a good idea. Just metal/plasteel alone makes it too easy to arm up the whole crew. Can't think of anything else off the top of my head, need more coffee

**Comment by shazbot194** [comment]
Update 29/1/18: (Same from the main post) Hello all, I have added two big features to this PR. The first is that the flare rifle can no longer shoot combat shells perfectly, exploding in your face when you do, about 40% of the time. This will shoot the shell at you, doing the full damage of the shell. But you can add some plasteel to stop this, but as normal, if there is a shell loaded when you add the sheets, you will shoot yourself with the shell. The description will change when you add the sheets, making it possible to tell if one is reinforced. The other feature is that semi-auto shotguns like the combat shotgun don't cycle themselves when loaded with flares. The bulldog still does, mostly because the code would be too much and possible drag down performance, so it will be allowed to be the only auto shotgun to fully be compatible with flares.

File: code/modules/crafting/guncrafting.dm [hunk 1]
```diff
@@ -81,3 +95,34 @@
 			to_chat(user, "<span class='warning'>You need at least five feet of wrapping paper to secure the stock.</span>")
 			return
 
+//Flare Rifle//
+
+/obj/item/weaponcrafting/flaregunconstruction
```
Comment (taukausanake) [inline code/modules/crafting/guncrafting.dm]: Just on the side, but I thought we got rid of all the step construction things in favor for table crafting. Were firearm weapons excluded from this?

Comment (shazbot194) [reply code/modules/crafting/guncrafting.dm]: Somehow they weren't gotten rid of, so I figured I would also add this including the hand crafting with the table crafting.

**Comment by Shadeykins** [comment]
I guess you could say you've got a real _flare_ for this kind of thing.

**Comment by Denthamos** [comment]
Hmmmn.  How many flares does it take to kill a monkey is now what I wonder.

**Comment by shazbot194** [comment]
Each flare does 10 burn damage, so the answer is too many.

**Comment by shazbot194** [comment]
Hmmm, merge conflicts, well, it would be nice to hear if there have been any updates on if this is getting in or not, or if talks are still on going.

**Comment by shazbot194** [comment]
Sigh, thanks merge conflicts, not fixing even after I fix them

**Comment by shazbot194** [comment]
Small update, I have removed slap crafting and changed the icon to match the new icons me and merchant are making.

**Comment by taukausanake** [comment]
I honestly do miss slap crafting. It make you feel like you were actually building something

**Comment by shazbot194** [comment]
I agree, but I know other people don't want it, and I don't want to go against the grain on something like this.

**Comment by shazbot194** [comment]
So, any update on this PR, besides me having to fix the code a little.

**Comment by tigercat2000** [comment]
The code for this is rather unwieldy and, while it could be mostly fixed, there are a few major issues, one of which is the fact that the ammo type spits out flashlight-type items. Additionally, this isn't really very useful in general, especially seeing as it's not much more than a pneumatic cannon full of flashlights.

As such, closing.

**Pull request closed without merging**